lib/widgets/channel_list.dart

```diff
@@ -34,12 +34,32 @@ class _ChannelListPageState extends State<ChannelListPage> with PerAccountStoreA
     return Scaffold(
       appBar: AppBar(title: Text(zulipLocalizations.channelListPageTitle)),
       body: SafeArea(
-        child: ListView.builder(
+        child: streams.isEmpty ? const _NoStreamsItem() : ListView.builder(
```

Comment:
This page is currently only opened through `SubscriptionListPage`, So if that page just doesn't show the button when there are no non-subscribed streams, this would only occur if `ChannelListPage` was already opened and new events on the store removed all the streams. So this is a good case to have.